src/npc.cpp

```diff
@@ -1783,6 +1787,30 @@ Creature::Attitude npc::attitude_to( const Creature &other ) const
     return A_NEUTRAL;
 }
 
+void npc::npc_dismount()
+{
+    tripoint pnt;
+    for( const auto &elem : g->m.points_in_radius( pos(), 1 ) ) {
+        if( g->is_empty( elem ) ) {
+            pnt = elem;
+            break;
+        }
+    }
+    if( pnt != tripoint_zero && mounted_creature ) {
+        remove_effect( effect_riding );
+        monster *critter = mounted_creature.get();
```

Comment:
Why make this a separate variable? You can continue to use `mounted_creature` until you set it to null.